doc/source/api/alleleAnnotations.rst

```diff
@@ -0,0 +1,84 @@
++---------------------+---------------------------------------------------------------------------------------------------------------------+
+| VariantAnnotation   | A VariantAnnotation record represents the result of comparing a single variant to the set of reference data. It     |
+|                     | contains structured sub-records and a flexible key-value pair ‘info’ field.                                         |
++---------------------+---------------------------------------------------------------------------------------------------------------------+
+| TranscriptEffect    | A transcript effect record describes the effect of an allele on a transcript.                                       |
++---------------------+---------------------------------------------------------------------------------------------------------------------+
+| AlleleLocation      | An allele location record holds the location of an allele relative to a non-genomic coordinate system such as a CDS |
+|                     | or protein. It holds the reference and alternate sequence where appropriate                                         |
++---------------------+---------------------------------------------------------------------------------------------------------------------+
+| AnalysisResult      | An AnalysisResult record holds the output of a prediction package such as SIFT on a specific allele.                |
++---------------------+---------------------------------------------------------------------------------------------------------------------+
+
+
+TranscriptEffect attributes
+@@@@@@@@@@@@@@@@@@@@@@@@@@@
+
+A ``VariantAnnotation`` record may have many ``TranscriptEffect`` records as one is 
+reported for each possible combination of alternate alleles and overlapping 
+transcripts. The record includes:
+
+* The identifier of the transcript feature the variant was analysed against.
+* The alternate allele of the variant analysed. This is necessary as the current variant model supports multiple alternate alleles.
+* The predicted effects of the allele on the transcript, which should be described using `Sequence Ontology <http://www.sequenceontology.org>`_ terms.
+* Human Genome Variation Society (`HGVS <http://www.hgvs.org/mutnomen>`_ variant description nomenclature at genomic, transcript and protein level. 
+* ``AlleleLocation`` records describing the changes at cDNA, CDS and protein level.
+* A set of results from prediction packages analyzing the allele impact.
+* A summary impact classification reflecting the highest impact consequence.
+
+Impact Classification
+@@@@@@@@@@@@@@@@@@@@@
+
+The predicted impact is a simple prioritization or classification based on the 
+putative deleteriousness of the variant allele on the transcript. This is usually 
+calculated based on naive algorithms and may not accurately predict true impact 
+at protein level. 
+
```

Comment:
> @diekhans Probably because there isn't any "vendor independent" version that
> works. HGVS doesn't really work, either; at least not for structural
> variations.

The reason for documenting rationale is so one knows what the
designers were thinking; not have to second guess ;-)